src/main/java/teammates/logic/core/CoursesLogic.java

```diff
@@ -782,4 +792,20 @@ public boolean isCourseArchived(CourseAttributes course, InstructorAttributes in
         }
         return result;
     }
+    
+    public List<String> getArchivedCourseIds(List<CourseDetailsBundle> allCourses, List<InstructorAttributes> instructorList) {
+        List<String> activeCourseIds = new ArrayList<String>();
+        for (CourseDetailsBundle cdb : allCourses) {
+            if (cdb.course.isArchived) {
```

Comment:
Right, we should. #3748, #3747 and #3511, was working on that but stopped to do jstl a few months ago. I'll finish those up in the next milestone.